sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -930,7 +930,7 @@ complexColTypeList
     ;
 
 complexColType
-    : identifier ':' dataType (NOT NULL)? commentSpec?
+    : identifier ':'? dataType (NOT NULL)? commentSpec?
```

Comment:
Could you update the SQL doc? https://github.com/apache/spark/blame/master/docs/sql-ref-datatypes.md#L197

Reply:
OK